File: c#/ChatAPI/ClientConnectionInfo.cs
namespace ChatAPI;

/// contains all information needed to connect to a client
public class ClientConnectionInfo
{
    /// version of ChatAPI implemented on the client
    public short apiVersion;
    /// client's public RSA encryption key
    public string publicKey;
    /// encryption algorithms implemented on the client
    public string[] encryptionAlgorithms;
    /// hash algorithms implemented on the client
    public string[] hashAlgorithms;

    public ClientConnectionInfo(short apiVersion, string publicKey, string[] encryptionAlgorithms, string[] hashAlgorithms)
    {
        this.apiVersion = apiVersion;
        this.publicKey = publicKey;
        this.encryptionAlgorithms = encryptionAlgorithms;
        this.hashAlgorithms = hashAlgorithms;
    }

    public override string ToString() =>
        new StringBuilder()
            .Append("ClientConnectionInfo: {\n")
            .Append("apiVersion: ").Append(apiVersion)
            .Append(";\npublicKey: ").Append(publicKey)
            .Append(";\nencryptionAlgorithms: ").Append(encryptionAlgorithms.MergeToString(','))
            .Append(";\nhashAlgorithms: ").Append(hashAlgorithms.MergeToString(','))
            .Append("\n}").ToString();
}

File: java/ChatAPI/src/chat/api/ClientConnectionInfo.java
package chat.api;

import java.security.MessageDigest;

public class ClientConnectionInfo {
    short apiVersion;
    String publicKey;
    String[] encryptionAlgorithms;
    String[] hashAlgorithms;

    public ClientConnectionInfo(short apiVersion, String publicKey, String[] encryptionAlgorithms, String[] hashAlgorithms) {
        this.apiVersion=apiVersion;
        this.publicKey=publicKey;
        this.encryptionAlgorithms=encryptionAlgorithms;
        this.hashAlgorithms=hashAlgorithms;
    }

    @Override
    public String toString(){
        var b=new StringBuilder()
                .append("ClientConnectionInfo: {\n")
                .append("apiVersion: ").append(apiVersion)
                .append(";\npublicKey: ").append(publicKey)
                .append(";\nencryptionAlgorithms: ");
        StringBuilderExtended.appendArray(b,encryptionAlgorithms, ',')
                .append(";\nhashAlgorithms: ");
        StringBuilderExtended.appendArray(b,hashAlgorithms, ',')
                .append("\n}");
        return b.toString();
    }
}
